storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,644 @@
+        String origWorkerId = _workerId;
+        super.cleanUpForRestart();
+        synchronized (_localState) {
+            Map<String, Integer> workersToPort = _localState.getApprovedWorkers();
+            workersToPort.remove(origWorkerId);
+            removeWorkersOn(workersToPort, _port);
+            _localState.setApprovedWorkers(workersToPort);
+            LOG.info("Removed Worker ID {}", origWorkerId);
+        }
+    }
+
+    @Override
+    public void relaunch() throws IOException {
+        _type.assertFull();
+        //We are launching it now...
+        _type = ContainerType.LAUNCH;
+        createNewWorkerId();
+        setup();
+        launch();
+    }
+
+    @Override
+    public boolean didMainProcessExit() {
+        return _exitedEarly;
+    }
+
+    /**
+     * Run the given command for profiling
+     * 
+     * @param command
+     *            the command to run
+     * @param env
+     *            the environment to run the command
+     * @param logPrefix
+     *            the prefix to include in the logs
+     * @param targetDir
+     *            the working directory to run the command in
+     * @return true if it ran successfully, else false
+     * @throws IOException
+     *             on any error
+     * @throws InterruptedException
+     *             if interrupted wile waiting for the process to exit.
+     */
+    protected boolean runProfilingCommand(List<String> command, Map<String, String> env, String logPrefix,
+            File targetDir) throws IOException, InterruptedException {
+        _type.assertFull();
+        Process p = SupervisorUtils.launchProcess(command, env, logPrefix, null, targetDir);
+        int ret = p.waitFor();
+        return ret == 0;
+    }
+
+    @Override
+    public boolean runProfiling(ProfileRequest request, boolean stop) throws IOException, InterruptedException {
+        _type.assertFull();
+        String targetDir = ConfigUtils.workerArtifactsRoot(_conf, _topologyId, _port);
+
+        @SuppressWarnings("unchecked")
+        Map<String, String> env = (Map<String, String>) _topoConf.get(Config.TOPOLOGY_ENVIRONMENT);
+        if (env == null) {
+            env = new HashMap<String, String>();
+        }
+
+        String str = ConfigUtils.workerArtifactsPidPath(_conf, _topologyId, _port);
+
+        String workerPid = _ops.slurpString(new File(str)).trim();
+
+        ProfileAction profileAction = request.get_action();
+        String logPrefix = "ProfilerAction process " + _topologyId + ":" + _port + " PROFILER_ACTION: " + profileAction
+                + " ";
+
+        List<String> command = mkProfileCommand(profileAction, stop, workerPid, targetDir);
+
+        File targetFile = new File(targetDir);
+        if (command.size() > 0) {
+            return runProfilingCommand(command, env, logPrefix, targetFile);
+        }
+        LOG.warn("PROFILING REQUEST NOT SUPPORTED {} IGNORED...", request);
+        return true;
```

Comment:
Oh right, that makes sense. I wasn't looking at the place this is used, just went by the Javadoc for the overridden method.